apps/web/app/(use-page-wrapper)/(main-nav)/ShellMainAppDir.tsx

```diff
@@ -0,0 +1,65 @@
+import { ShellMainAppDirBackButton } from "app/(use-page-wrapper)/(main-nav)/ShellMainAppDirBackButton";
+import classNames from "classnames";
+
+import type { LayoutProps } from "@calcom/features/shell/Shell";
+
+// Copied from `ShellMain` but with a different `ShellMainAppDirBackButton` import
+// for client/server component separation
+export function ShellMainAppDir(props: LayoutProps) {
```

Comment:
Copied from `ShellMain` component inside `Shell.tsx`.
I was able to turn this into RSC by extracting the back button into a separate client component `ShellMainAppDirBackButton.tsx`.